plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java

```diff
@@ -0,0 +1,336 @@
+                EntitySearchFilter.DESC_ORDER : EntitySearchFilter.ASC_ORDER);
+
+        return filter;
+    }
+
+    //TODO support array of categories
+    private List<String> extractCategoriesFromFilters(RestListRequest requestList) {
+        List<String> categories = new ArrayList<>();
+
+        for (Filter filter : Optional.ofNullable(requestList.getFilters()).orElse(new Filter[]{})) {
+            if (filter.getAttribute().equals("categories")) {
+                categories.add(filter.getValue());
+            }
+        }
+
+        return categories;
+    }
+
+    private AssetDto convertResourceToDto(String resourceType, ResourceInterface resource) {
+        if ("Image".equals(resourceType)) {
+            return convertImageResourceToDto((ImageResource) resource);
+        } else if ("Attach".equals(resourceType)) {
+            return convertFileResourceToDto((AttachResource) resource);
+        } else {
+            throw new RestServerError(String.format("Invalid resource type: %s", resourceType), null);
+        }
+    }
+
+    //TODO better encapsulate dto conversion, perhaps create separate builder?
+    private ImageAssetDto convertImageResourceToDto(ImageResource resource) {
+        ImageAssetDto.ImageAssetDtoBuilder builder = ImageAssetDto.builder()
+                .id(resource.getId())
+                .name(resource.getMasterFileName())
+                .description(resource.getDescription())
+                .createdAt(resource.getCreationDate())
+                .updatedAt(resource.getLastModified())
+                .group(resource.getMainGroup())
+                .metadata(resource.getMetadata())
+                .version(ImageMetadataDto.builder()
+                        .path(resource.getImagePath("0"))
+                        .size(resource.getDefaultInstance().getFileLength())
+                        .build());
+
+        for (Category category : resource.getCategories()) {
+            builder.category(category.getCode());
+        }
+
+        for (ImageResourceDimension dimensions : getImageDimensions()) {
+            ResourceInstance instance = resource.getInstance(dimensions.getIdDim(), null);
+
+            if (instance == null) {
+                log.warn("ResourceInstance not found for dimensions id {} and image id {}", dimensions.getIdDim(), resource.getId());
+                continue;
+            }
+
+            builder.version(ImageMetadataDto.builder()
+                    .path(resource.getImagePath(String.valueOf(dimensions.getIdDim())))
+                    .size(instance.getFileLength())
+                    .dimensions(String.format("%dx%d px", dimensions.getDimx(), dimensions.getDimy()))
+                    .build());
+        }
+
+        return builder.build();
+    }
+
+    private FileAssetDto convertFileResourceToDto(AttachResource resource){
+        FileAssetDto.FileAssetDtoBuilder builder = FileAssetDto.builder()
```

Comment:
Very nice!

Reply:
Lombok FTW!
Can still be better with streams, pushing changes now.